bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+        if isinstance(x, str):
+            #Can be string constant or can be column name
+            if x in self.col_names: # If x is a column
+                t = self.rdf.GetColumnType(x)
+                if t in map_tree_to_numbadeclares.keys(): # The column is a fundemental type from tree  
+                    return map_tree_to_numbadeclares[t]
+                elif '<' in  t: # The column type is a RVec<type>
+                    g = re.match('(.*)<(.*)>', t).groups(0)
+                    if g[1] in map_tree_to_numbadeclares.keys():
+                        return "RVec<" + map_tree_to_numbadeclares[g[1]] + ">"
+                    return "RVec<" + str(g[1]) + ">" # There are data type that leak into here. NOt sure from where. BUt need to implement something here.
+                    
+                else:
+                    return t
+            else :
+                return 'str'
+                #! Numba Declare does not support "string" type. Check _numbadeclare.Thus, Cannot pass string constants to the filter/Defines..
+        elif isinstance(x, bool):
+            return 'bool'
+        elif isinstance(x, int):
+            return 'int'
+        elif isinstance(x, float):
+            return 'float'
+        elif isinstance(x, np.ndarray): 
+            if x.dtype.type in  map_numpy_to_tree:
+                return "RVec<" + map_numpy_to_tree[x.dtype.type] + ">"
+            else:
+                raise TypeError(f"Support for {x.dtype.type} arrays is not yet supported.")
+        #! Need to work out how to map things like tuples, dicts, lists...
+        else:
+                raise TypeError(f"Type of {type(x)}:  {x} is not supported to be jitted.")
+            
+    def find_function_signature(self, func, funcargs):
+        """
+        Function to determine the function signature i.e the type of variables the function arguments and its corresponding value.
+        Args:
+	    `   func: Function whose signature is to be determined.
+                    P.S The function should not be crazy. It should have simple datatypes for the arguments and should not have default arguments. (Use funcargs instead)
+            funcargs: Dictionatry containing constant parameters the that func requires. Or any special mapping of the arguments of func.
+
+        Returns:
+            input_args: Map of func's input variables to values
+            type_args : Map of func'c input variables to its corresponding type.
+        """
+        try:
+            import inspect 
+        except:
+            raise ImportError("Failed to import inspect during call to determine function signature.")
+        func_sign = inspect.signature(func)
+        # Find the Return type
+        if func_sign.return_annotation is inspect._empty:
+            raise ValueError("Return type of the function is not mentioned.\n Function cannot be jitted as signature cannot be determined.")
```

Comment:
This error is confusing for two reasons:
1. In this PR you are only dealing with `Filter`, so you always know the return type.
2. Even considering this class will be reused for the `Define`, we don't want to error out if the user doesn't provide the return type. Instead, we want to infer it for them, so this error should not exist

Reply:
I guess we should check that Numba inferred the return type of the Filter correctly / the user did the right thing in their function? In other words, that the return type is bool?